piecrust/rendering.py

```diff
@@ -32,6 +32,7 @@ def __init__(self, page, route, route_metadata):
         self.page = page
         self.route = route
         self.route_metadata = route_metadata
+        self.pretty_urls = page.config.get("pretty_urls")
```

Comment:
I don't think you need to cache this here in the `QualifiedPage` class, the `PageBaker` can access it itself.